pages/add-device.js

```diff
@@ -3,22 +3,95 @@ import Link from "next/link";
 import { useState } from "react";
 import AddForm from "@/components/AddDeviceForm";
 import Card from "@/components/Card";
+import Doughnut from "@/components/DoughnutChart";
 
 export default function AddNewDevice({ createDevice, devices, handleDelete }) {
   const [toggleForm, setToggleForm] = useState(false);
+  const [overallCost, setOverallCost] = useState(0);
   const sortedDevices = [...devices];
 
   function handleSubmit(event) {
     event.preventDefault();
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
-    createDevice(data);
+    setOverallCost((sumUpDevices / 1000) * data.price);
     event.target.reset();
   }
 
+  function filterByCategory(category) {
+    return (device) => device.device_category === category;
+  }
+
+  const calculateDailyPowerConsumption = (device) =>
+    device.power_consumption * device.average_usage_time;
+
+  const sum = (accumulator, currentValue) => accumulator + currentValue;
+
+  const sumUpDevices = devices
+    .map(calculateDailyPowerConsumption)
+    .reduce(sum, 0);
+
+  const sumUpAppliances = devices
+    .filter(filterByCategory("Appliances"))
+    .map(calculateDailyPowerConsumption)
+    .reduce(sum, 0);
+
+  const sumUpEntertainment = devices
+    .filter(filterByCategory("Entertainment"))
+    .map(calculateDailyPowerConsumption)
+    .reduce(sum, 0);
+
+  const sumUpWork = devices
+    .filter(filterByCategory("Work"))
+    .map(calculateDailyPowerConsumption)
+    .reduce(sum, 0);
+
+  const sumUpLighting = devices
+    .filter(filterByCategory("Lighting"))
+    .map(calculateDailyPowerConsumption)
+    .reduce(sum, 0);
+
+  const chartData = {
+    labels: ["Entertainment", "Appliances", "Work", "Lighting"],
+    datasets: [
+      {
+        label: "Power consumption",
+        data: [sumUpEntertainment, sumUpAppliances, sumUpWork, sumUpLighting],
+        backgroundColor: [
+          "rgba(255, 99, 132, 0.5)",
+          "rgba(54, 162, 235, 0.5)",
+          "rgba(255, 206, 86, 0.5)",
+          "rgba(75, 192, 192, 0.5)",
+        ],
+        borderColor: [
+          "rgba(255, 99, 132, 1)",
+          "rgba(54, 162, 235, 1)",
+          "rgba(255, 206, 86, 1)",
+          "rgba(75, 192, 192, 1)",
+        ],
+        borderWidth: 1,
+      },
+    ],
+  };
   return (
     <>
       <Link href="/">Home</Link>
+      <Doughnut chartData={chartData} />
+      <h2> {overallCost.toFixed(2)} $</h2>
```

Comment:
Hi Johannes,
this might help you with the currency format.
```javascript
// const overallCost = 123456.789;
<h2>{new Intl.NumberFormat('de-DE', { style: 'currency', currency: 'EUR' }).format(overallCost)}</h2>
// Expected output: "123.456,79 €"
```

Reply:
thanks for your comment 👍